### PR TITLE
[Feature] [Louie] Specific Product Page conditional back function

### PR DESCRIPTION
**Pull request by cezro**
**Clickup Task:**
https://app.clickup.com/t/865cr0mve

**Description:**
fix for navigation for back button on the specific product page

**Images/Videos:**
https://github.com/ezerssss/unihub/assets/90561528/64efd96a-337d-4262-8001-08d2f85a7b6d

### CODE REVIEW

File: screens/SpecificProduct/SpecificProduct.tsx [hunk 1]
```diff
@@ -105,7 +104,16 @@ function SpecificProduct({ route, navigation }: ProductNavigationProps) {
     );
   }
 
-  const goBack = useGoBack();
+  function goBack() {
+    const routes = navigation.getState().routes;
+    const prevRoute = routes[routes.length - 2];
+
+    if (prevRoute.name === Routes.SELL) {
+      navigation.navigate(Routes.HOME);
```
Comment (ezerssss) [inline screens/SpecificProduct/SpecificProduct.tsx:112]: do we want to redirect to home directly?

how about the scenario:

- user goes to a category page
- user navigates to sell screen
- user posts a product
- user is redirected to the specific product page

when he/she presses back should he/she be redirected to home or back to the category page?

another scenario:

- user navigates to category
- user clicks one of his/her products
- user edits and navigates to edit screen
- user edited the product and is redirected to the specific product page

when he/she presses back should he/she be redirected to home or back to category page?

**Review by ezerssss — CHANGES_REQUESTED** [review]

File: screens/SpecificProduct/SpecificProduct.tsx [hunk 2]
```diff
@@ -105,7 +104,16 @@ function SpecificProduct({ route, navigation }: ProductNavigationProps) {
     );
   }
 
-  const goBack = useGoBack();
+  function goBack() {
+    const routes = navigation.getState().routes;
+    const prevRoute = routes[routes.length - 2];
```
[the remark this answers is not in the document]
Comment (cezro) [reply screens/SpecificProduct/SpecificProduct.tsx:109]: hmmm, i'm guessing this is for the notification redirect? Only case I could think of where routes would be less than 2 length

also wouldn't `.length` return the object length starting from 1 where the indexing using `[]` starts from 0?

[the remark this answers is not in the document]
Comment (ezerssss) [reply screens/SpecificProduct/SpecificProduct.tsx:109]: ah yes ur right, sorry bosss

**Review by ezerssss — APPROVED** [review]

**Review by jedmamosto — COMMENTED** [review]
so far so good, let's just wait on the footer ui to be merged so we can test with user page

**Pull request merged**